modules/mono/glue/GodotSharp/GodotSharp/Core/Basis.cs

```diff
@@ -267,45 +282,253 @@ public real_t Determinant()
         }
 
         /// <summary>
-        /// Returns the basis's rotation in the form of Euler angles
-        /// (in the YXZ convention: when *decomposing*, first Z, then X, and Y last).
-        /// The returned vector contains the rotation angles in
-        /// the format (X angle, Y angle, Z angle).
+        /// Returns the basis's rotation in the form of Euler angles.
+        /// The Euler order depends on the [param order] parameter,
+        /// for example using the YXZ convention: when decomposing,
+        /// first Z, then X, and Y last. The returned vector contains
+        /// the rotation angles in the format (X angle, Y angle, Z angle).
         ///
-        /// Consider using the <see cref="Quat()"/> method instead, which
+        /// Consider using the <see cref="RotationQuat"/> method instead, which
         /// returns a <see cref="Godot.Quat"/> quaternion instead of Euler angles.
         /// </summary>
+        /// <param name="order">The Euler order to use. By default, use YXZ order (most common).</param>
         /// <returns>A <see cref="Vector3"/> representing the basis rotation in Euler angles.</returns>
-        public Vector3 GetEuler()
+        public Vector3 GetEuler(EulerOrder order = EulerOrder.YXZ)
```

Comment:
This breaks binary compatibility, not sure if we care about that in 3.x.

Also, note that GDScript doesn't seem to expose a parameter in the `get_euler` method so if we are backporting this I suppose it should be done in `basis.h` too.

Reply:
We could make the argument optional and then make an argless version that calls `GetEuler(EulerOrder.YXZ)` if that's preferred. I'm not sure if ABI compat is necessary in C#, since I think users will always be recompiling their projects when moving from 3.5 to 3.6.

Reply:
It's usually a problem with libraries because they reference an older version of GodotSharp but their project still compiles, it only fails on runtime. It has happened before, see https://github.com/godotengine/godot/issues/64286.

Reply:
Updated as requested, should preserve ABI compatibility now.